pyls/python_ls.py

```diff
@@ -39,8 +40,10 @@ def capabilities(self):
             'textDocumentSync': lsp.TextDocumentSyncKind.INCREMENTAL
         }
 
-    def initialize(self, root_path, init_opts, _process_id):
-        self.workspace = Workspace(root_path, lang_server=self)
+    def initialize(self, root_uri, init_opts, _process_id):
+        self.workspace = Workspace(root_uri, lang_server=self)
+
+        root_path = unquote(urlparse(root_uri).path)
         self.config = config.Config(root_path, init_opts)
```

Comment:
Do you mean Config should accept `root_uri` and convert it to `self.root_path`?

I agree we should convert uri -> path only once, but thought Workspace is a good place to do it as it receives both URIs and contains filesystem logic. 

Related: Workspace has an `is_local` method which is only used in a test. If we remove this function, Workspace also has no need for its `_url_parsed` field.